### PR TITLE
Local Complex Interaction

### PR DESCRIPTION
**Pull request by ChenxiZhaoTUM**
This is the first case to implement multiple boundary conditions.

### CODE REVIEW

**Review by Bo-Zhang1995 — APPROVED** [review]

**Comment by Xiangyu-Hu** [comment]
@ChenxiZhaoTUM  building is failed.

**Review by Xiangyu-Hu — CHANGES_REQUESTED** [review]

File: tests/user_examples/ExtraSources/extra_sources_share/diffusion_reaction_particles_with_boundary.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/**
```
[the remark this answers is not in the document]
Comment (Xiangyu-Hu) [reply tests/user_examples/ExtraSources/extra_sources_share/diffusion_reaction_particles_with_boundary.h:1]: These particles definitions should be deleted.

**Comment by Xiangyu-Hu** [comment]
@ChenxiZhaoTUM You need compile and run Ctest locally first and, after all tests passed, then initialize a CI test on Github by commit your changes. Otherwise github will waste the computation time for unsuccessful compiling and testing.

**Comment by ChenxiZhaoTUM** [comment]
> @ChenxiZhaoTUM You need compile and run Ctest locally first and, after all tests passed, then initialize a CI test on Github by commit your changes. Otherwise github will waste the computation time for unsuccessful compiling and testing.

sorry about that, I thought I had closed this pull request. Now the local ctest has been passed, and I am retesting it by CI.

**Comment by Xiangyu-Hu** [comment]
@ChenxiZhaoTUM you need merge master to your branch first, as the master has changed the directory structure.
There should be should be no conflict for it.

**Pull request merged**